### PR TITLE
fix: sync PRs when chainPR was changed

### PR DESCRIPTION
**Pull request by yuichi10**
## Context
When change chainPR of pipeline, UI does not change automatically.
This PR sync PRs and change UI automatically when chainPR was changed.

## Objective
- syncPRs function call syncPR functions. 
   To sync chainPR's status, we need to get pipeline configuration to check next job of workflowParser. Now syncPRs function does not see pipeline configuration. If syncPRs get pipeline configuration, there are almost no difference to call syncPR function.

- syncPRs when chainPR was changed
  Maybe it took many time to sync PRs, because it get pipeline configuration and check next job. Thus I call syncPRs function only chainPR was changed.

### CODE REVIEW

File: lib/pipeline.js [hunk 1]
```diff
@@ -441,14 +361,17 @@ class PipelineModel extends BaseModel {
...
-                ]);
-            }));
+                openedPRs.forEach((openedPR) => {
+                    const prNum = openedPR.name.split('-')[1];
+
+                    synced.push(this.syncPR(prNum));
+                });
+
+                return Promise.all(synced)
+                    .then(() => this._archiveClosePRs(existingJobs, openedPRs));
```
Comment (sakka2) [inline lib/pipeline.js:373]: [ask] why do we need to call this function after finishing syncPR? It seems that closed PR job will be archived in syncPR function.

Comment (yuichi10) [reply lib/pipeline.js:373]: To archive jobs which are in closed PR.
SyncPR function does not have this feature, thus I add this function.

Comment (sakka2) [reply lib/pipeline.js:373]: 👍

**Review by sakka2 — APPROVED** [review]

File: lib/eventFactory.js [hunk 2]
```diff
@@ -425,10 +425,12 @@ class EventFactory extends BaseFactory {
             pr: {},
             prNum
         };
+        let beforeChainPR = '';
```
Comment (s-yoshika) [inline lib/eventFactory.js]: [imo] I prefer prevChainPR.

**Review by s-yoshika — APPROVED** [review]
Add a small comment, but LGTM.

File: lib/eventFactory.js (hunk 2 above)
Comment (yuichi10) [reply lib/eventFactory.js]: I will fix that.

**Review by jithine — APPROVED** [review]

**Review by minzcmu — APPROVED** [review]

**Comment by yuichi10** [comment]
@minz1027 
Thank you!!!

**Pull request merged**